src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CamundaServiceTest.java

```diff
@@ -1085,5 +1086,137 @@ void should_auto_complete_and_return_empty_array() {
+        @Test
+        void cancelTask_should_throw_insufficient_permissions_exception_when_user_did_not_have_enough_permission() {
+
+            String taskId = UUID.randomUUID().toString();
+            Assignment mockedRoleAssignment = mock(Assignment.class);
+            Map<String, CamundaVariable> mockedVariables = mockVariables();
+
+            UserInfo mockedUserInfo = mock(UserInfo.class);
+            when(mockedUserInfo.getUid()).thenReturn(IDAM_USER_ID);
+
+            List<PermissionTypes> permissionsRequired = asList(CANCEL);
+            AccessControlResponse accessControlResponse =
+                new AccessControlResponse(mockedUserInfo, singletonList(mockedRoleAssignment));
+
+            when(camundaServiceApi.getVariables(BEARER_SERVICE_TOKEN, taskId)).thenReturn(mockedVariables);
+
+            when(permissionEvaluatorService.hasAccess(
+                mockedVariables,
+                singletonList(mockedRoleAssignment),
+                permissionsRequired
+            )).thenReturn(false);
+
+
+            assertThatThrownBy(() -> camundaService.cancelTask(taskId, accessControlResponse, permissionsRequired))
+                .isInstanceOf(InsufficientPermissionsException.class)
+                .hasNoCause()
+                .hasMessage(
+                    String.format("User did not have sufficient permissions to cancel task with id: %s", taskId));
+        }
+
+        @Test
+        void cancelTask_should_throw_a_server_error_exception_when_completing_task_fails() {
+
+            String taskId = UUID.randomUUID().toString();
+            Assignment mockedRoleAssignment = mock(Assignment.class);
+            Map<String, CamundaVariable> mockedVariables = mockVariables();
+
+            UserInfo mockedUserInfo = mock(UserInfo.class);
+            when(mockedUserInfo.getUid()).thenReturn(IDAM_USER_ID);
+
+            AccessControlResponse accessControlResponse = new AccessControlResponse(
+                mockedUserInfo, singletonList(mockedRoleAssignment)
+            );
+
+            List<PermissionTypes> permissionsRequired = singletonList(CANCEL);
+
+            when(camundaServiceApi.getVariables(BEARER_SERVICE_TOKEN, taskId)).thenReturn(mockedVariables);
+
+            when(permissionEvaluatorService.hasAccess(
+                mockedVariables,
+                accessControlResponse.getRoleAssignments(),
+                permissionsRequired
+            )).thenReturn(true);
+
+            doThrow(mock(FeignException.class))
+                .when(camundaServiceApi).bpmnEscalation(any(), any(), anyMap());
+
+            assertThatThrownBy(() -> camundaService.cancelTask(taskId, accessControlResponse, permissionsRequired))
+                .isInstanceOf(ServerErrorException.class)
+                .hasCauseInstanceOf(FeignException.class)
+                .hasMessage(String.format(
+                    "There was a problem cancelling the task with id: %s",
+                    taskId));
+
+        }
+
+    }
+
```

Comment:
do we need a test scenario for what happens when a task we are trying to cancel does not exist?

Reply:
Its covered in FT, have a look at 404 test.